evap/student/views.py

```diff
@@ -48,15 +53,34 @@ def index(request):
     semester_list = [dict(
         semester_name=semester.name,
         id=semester.id,
-        is_active=semester.is_active,
         results_are_archived=semester.results_are_archived,
         grade_documents_are_deleted=semester.grade_documents_are_deleted,
         evaluations=[evaluation for evaluation in evaluations if evaluation.course.semester_id == semester.id]
     ) for semester in semesters]
 
+    unfinished_evaluations = list(
+        Evaluation.objects
+        .filter(participants=request.user, state__in=['prepared', 'editor_approved', 'approved', 'in_evaluation'])
+        .exclude(voters=request.user)
+        .prefetch_related('course__responsibles', 'course__type', 'course__semester', 'participants')
+    )
+
+    # available evaluations come first, ordered by time left for evaluation and the name
+    # evaluations in other (visible) states follow by name
+    def sorter(evaluation):
+        return (
+            evaluation.state != 'in_evaluation',
+            evaluation.state not in ['prepared', 'editor_approved', 'approved'],
+            date_to_datetime(evaluation.vote_end_date) - datetime.now() if evaluation.state == 'in_evaluation' else evaluation.full_name,
+            evaluation.full_name
```

Comment:
Maybe i’m not quite understanding key functions in Python, but if my considerations are correct, then I have some points:

- `unfinished_evaluations` only have the states `['prepared', 'editor_approved', 'approved', 'in_evaluation']`, so in my head the second line is superflous. If you want to protect yourself for the situation when more states are added, I’d prefer an explicit switch
  ```python
   if state == "in_evaluation": 0
   elif state in ['prepared', 'editor_approved', 'approved']: 1
   else: 2`
  ```
- Wouldn’t just `evaluation.vote_end_date` suffice here instead of subtracting `datetime.now()`? As comparable, it shouldn’t make a difference to ordering if you subtract a constant value
- Also, adding either a `datetime` or a `str` in the second last line, I’d prefer to add a null-value here (maybe `None` works, or a fixed date)